include/libultra/xprintf.h

```diff
@@ -0,0 +1,16 @@
+#ifndef XRINTF_H
+#define XRINTF_H
+
+#include "types.h"
+#include "va_args.h"
+
+#ifdef __cplusplus
+extern "C"{
+#endif
+
+extern int _Printf(void* (*prout_func)(void*, const char*, int), char* buf, const char* fmt, va_list ap);
```

Comment:
doesn't compile with this, you have it different in your scratch though https://decomp.me/scratch/SwiKT